dataikuapi/dss/ml.py

```diff
@@ -308,50 +331,554 @@ def set_metric(self, metric=None, custom_metric=None, custom_metric_greater_is_b
         self.mltask_settings["modeling"]["metrics"]["customEvaluationMetricNeedsProba"] = custom_metric_use_probas
 
+    def get_hyperparameter_search_settings(self):
+        return self.mltask_settings["modeling"]["gridSearchParams"]
+
     def save(self):
         """Saves back these settings to the ML Task"""
 
         self.client._perform_empty(
                 "POST", "/projects/%s/models/lab/%s/%s/settings" % (self.project_key, self.analysis_id, self.mltask_id),
                 body = self.mltask_settings)
 
+
+class HyperparameterSettings(object):
+
+    def __init__(self, name, algo_settings):
+        self.name = name
+        self._algo_settings = algo_settings
+
+
+class NumericalHyperparameterSettings(HyperparameterSettings):
+
+    def __repr__(self):
+        clean_dict = self._algo_settings._get_active_numerical_hyperparam_dict(self.name)
+        return self.__class__.__name__ + "\nhyperparameter: \"{}\"".format(self.name) + "\nsettings: " + json.dumps(clean_dict, indent=4)
+
+    __str__ = __repr__
+
+    def set_explicit_values(self, values, mode_as_explicit=True):
+        self._algo_settings._set_numerical_explicit_values(self.name, values, mode_as_explicit=mode_as_explicit)
+
+    def set_range(self, range_min=None, range_max=None, nb_values=None, mode_as_range=True):
+        self._algo_settings._set_numerical_range(self.name, range_min=range_min, range_max=range_max, nb_values=nb_values,
+                                                 mode_as_range=mode_as_range)
+
+    def set_mode_as_explicit(self):
+        self._algo_settings._set_hyperparameter_mode_as_explicit(self.name)
+
+    def set_mode_as_range(self):
+        self._algo_settings._set_hyperparameter_mode_as_range(self.name)
+
+
+class CategoricalHyperparameterSettings(HyperparameterSettings):
+
+    def __repr__(self):
+        return self.__class__.__name__ + "\nhyperparameter: \"{}\"".format(self.name) + "\nsettings: " + json.dumps(self._algo_settings[self.name], indent=4)
+
+    __str__ = __repr__
+
+    def set_values(self, values=None):
+        self._algo_settings._set_categorical_values(self.name, values=values)
+
+    def enable_values(self, values):
+        for value in values:
+            assert is_str_or_unicode(value)
+            self.set_values({value: {"enabled": True}})
+
+    def disable_values(self, values):
+        for value in values:
+            assert is_str_or_unicode(value)
+            self.set_values({value: {"enabled": False}})
+
+
+class SingleValuedHyperparameterSettings(HyperparameterSettings):
+
+    def __init__(self, name, algo_settings, accepted_values=None):
+        super(SingleValuedHyperparameterSettings, self).__init__(name, algo_settings)
+        self.accepted_values = accepted_values
+
+    def __repr__(self):
+        return self.__class__.__name__ + "\nhyperparameter: \"{}\"".format(self.name) + "\nvalue: " + json.dumps(self._algo_settings[self.name], indent=4)
```

Comment:
```
SingleValuedHyperparameterSettings
hyperparameter: "multi_class"
value: "ovr"
```

Maybe we could instead have a single line?

```
SingleValuedHyperparameterSettings(hyperparameter: "multi_class", value: "ovr")
```

And for multi-line settings, go from
```
CategoricalHyperparameterSettings
hyperparameter: "penalty"
settings: {
    "values": {
        "l1": {
            "enabled": true
        },
        "l2": {
            "enabled": true
        }
    }
}
```

To:

```
CategoricalHyperparameterSettings(hyperparameter: "penalty", settings: {
    "values": {
        "l1": {
            "enabled": true
        },
        "l2": {
            "enabled": true
        }
    }
})
```